public/app/plugins/panel/table/TablePanel.tsx

```diff
@@ -91,21 +114,188 @@ export class TablePanel extends Component<Props> {
+        return;
+      });
+      return { cells };
+    }
+
+    return { cells: undefined };
+  }
+
+  // get the footer cells from data provided by another frame
+  getProvidedFooter(frame: DataFrame): Footer {
+    const { options, data } = this.props;
+
+    const summaryFrame = data.series.find((f) => f.name === options.footerFrame);
+    if (summaryFrame === undefined) {
+      return { cells: undefined };
+    }
+
+    // group cells by field for multi value cells
+    const cellsByField: KeyValue = {};
+    frame.fields.forEach((f, i) => {
+      for (const sf of summaryFrame!.fields) {
+        const referenceField: string = sf.config?.custom?.['referenceField'];
+        const reducer: string = sf.config?.custom?.['reducer'];
+        if (referenceField === f.name) {
+          const cell = this.getSummaryCell(i, reducer, sf, true);
+          const fieldCells = cellsByField[referenceField] || [];
+          fieldCells.push(cell);
+          cellsByField[referenceField] = fieldCells;
+          continue;
+        }
+        continue;
+      }
+    });
+
+    let height: number | undefined;
+    const cells = frame.fields.map((f, i) => {
+      const fieldCells = cellsByField[f.name];
+      if (fieldCells === undefined) {
+        if (i === 0) {
+          // set the first col val to "Total" if not set
+          return 'Total';
+        }
+        return;
+      }
+      if (fieldCells.length > 1) {
+        const expandedHeight = fieldCells.length * 27;
+        height = height || 0;
+        if (height < expandedHeight) {
+          height = expandedHeight;
+        }
+      }
+      return this.getProvidedList(fieldCells);
+    });
+
+    return { cells, height };
+  }
+
+  getProvidedList(summaries: ReactNode[]) {
+    return (
+      <ul style={{ listStyle: 'none', width: '100%' }}>
+        {summaries.map((s: ReactNode, idx: number) => {
+          return <li key={idx}>{s}</li>;
+        })}
+      </ul>
+    );
+  }
+
+  getSummaryList(summaryProps: DynamicConfigValue[], fieldIndex: number, f: Field) {
+    const summaries = summaryProps.map((o: any) => {
+      return this.calculateSummaryCell(fieldIndex, o.value, f, false);
+    });
+    return (
+      <ul style={{ listStyle: 'none', width: '100%' }}>
+        {summaries.map((s: ReactNode, idx: number) => {
+          return <li key={idx}>{s}</li>;
+        })}
+      </ul>
+    );
+  }
+
+  getSummaryCell(i: number, functionName: string, f: Field, showLabel: boolean) {
+    const formatted = this.format(f.values.get(0), f);
+    return <SummaryCell key={i} func={functionName} value={formatted} showLabel={showLabel}></SummaryCell>;
+  }
+
+  calculateSummaryCell(i: number, functionName: string, f: Field, showLabel: boolean) {
+    if (functionName === '') {
+      return undefined;
+    }
+
+    if (functionName && functionName !== 'avg') {
+      const kv = this as KeyValue;
+      const func: Function = kv[functionName];
+      const val = this.calculate(f.values, func);
```

Comment:
Think using the getFieldDisplayValues should do the same job especially that the calcs may already be pre-calculated in the field.state. Similar thing is done i.e. in the Stat panel https://github.com/grafana/grafana/blob/073c84179fa0142732e10e21e87a0e5c74ddfd6b/public/app/plugins/panel/singlestat2/SingleStatPanel.tsx#L29

The formatting below should also be covered by getFieldDisplayValues as it takes field config into consideration.